### DISCUSSION TITLE
How to remove sectors with CommitWait PreCommitBatchWait and PreCommitBatchWait state?

### DISCUSSION THREAD

**Discussion by potato1992** [filecoin-project/lotus#7576] [Storage Provider] [answered]
How to remove sectors with CommitWait PreCommitBatchWait and PreCommitBatchWait state?
```
ID    State               OnChain  Active  Deals  
3825  CommitWait          NO       NO      CC     
3827  CommitWait          NO       NO      CC     
3828  CommitWait          NO       NO      CC     
3829  CommitWait          NO       NO      CC     
3830  CommitWait          NO       NO      CC     
3831  CommitWait          NO       NO      CC     
3832  CommitWait          NO       NO      CC     
3833  PreCommitBatchWait  NO       NO      CC     
3834  CommitWait          NO       NO      CC     
3835  CommitWait          NO       NO      CC     
3836  CommitWait          NO       NO      CC     
3837  CommitWait          NO       NO      CC     
3838  CommitWait          NO       NO      CC     
3839  CommitWait          NO       NO      CC     
3840  PreCommitBatchWait  NO       NO      CC     
3841  PreCommitBatchWait  NO       NO      CC     
3842  PreCommitBatchWait  NO       NO      CC     
3843  PreCommitBatchWait  NO       NO      CC     
3844  PreCommitBatchWait  NO       NO      CC     
3845  PreCommitBatchWait  NO       NO      CC     
3846  PreCommitBatchWait  NO       NO      CC     
3847  PreCommitBatchWait  NO       NO      CC     
3848  PreCommitBatchWait  NO       NO      CC     
3849  PreCommitBatchWait  NO       NO      CC     
3850  PreCommitBatchWait  NO       NO      CC    
```
`lotus-miner sectors update-state --really-do-it` does not work to change the state, and `lotus-miner sectors remove --really-do-it` does not work either.

By the way, those sectors are caused by not upgrading to v1.13.0 from v1.12. I really hope that the official take those thing seriously. When a Chia blockchain version is about to expire, the GUI/terminal will give out information long time before it really happens. But the Filecoin just upgrade it without any action to minimize miners' loss.

**Comment by de-authority** [comment, marked as answer]
> But the Filecoin just upgrade it without any action to minimize miners' loss.

It's not true. The team started notifying the community on the slack channel 14 days before the mandatory upgrade. It's true that some members of the community don't follow the updates in slack, so if you have a good idea, tell it here and help the team make it better!
And for your question. you can see this discussion about how to delete faulty sectors:https://github.com/filecoin-project/lotus/discussions/6950

**Reply by de-authority** [reply]
And you can try `lotus-shed sectors terminate <sector-num>`  before you remove them.

**Reply by potato1992** [reply]
> And you can try `lotus-shed sectors terminate <sector-num>` before you remove them.

I have tried, it does nothing. Do anyone know how to forcefully edit the database so I can remove them from the DB?

**Reply by de-authority** [reply]
TBH. I don't think we should edit the DB of lotus. I believe it's not safe though I am not a developer. 
if you can't terminate the sector. I think that's ok. because that's another chain's record.  
1: stop the sealing jobs if you still have sealing tasks in these sectors.
2: try to restart your miner.

**Comment by de-authority** [comment]
And can you tell me where do you check our mandatory upgrade information?

**Reply by potato1992** [reply]
> And can you tell me where do you check our mandatory upgrade information?

Thank you for your information, I read the upgrade information from the lotus release log only when I found my node stop sync.... Is there any way else I can get the upgrade information?

**Reply by de-authority** [reply]
1: Join our slack channel here: https://filecoinproject.slack.com/archives/C027TQMUVJN and here: https://filecoinproject.slack.com/archives/C01AC6999KQ  and you can register if you never join slack [here](https://filecoin.io/slack/). 
2: you can pay attention  to the announcement we post in GitHub forum [here](https://github.com/filecoin-project/lotus/discussions/categories/announcement)

**Reply by potato1992** [reply]
> 1: Join our slack channel here: https://filecoinproject.slack.com/archives/C027TQMUVJN and here: https://filecoinproject.slack.com/archives/C01AC6999KQ and you can register if you never join slack [here](https://filecoin.io/slack/). 2: you can pay attention to the announcement we post in GitHub forum [here](https://github.com/filecoin-project/lotus/discussions/categories/announcement)

Thank you!
